aten/src/ATen/native/TensorShape.cpp

```diff
@@ -68,7 +68,6 @@ Tensor& set_tensor_(Tensor& result, const Tensor& source) {
 Tensor& set_cpu_(Tensor& result) {
   Storage storage(
       Storage::use_byte_size_t(),
-      result.dtype(),
       0,
       c10::GetAllocator(kCPU),
       true);
```

Comment:
This function seems like it could change dtype. Safer to add an assert that `result.dtype() == dtype()` to make sure it doesn't.